tools/accuracy_checker/accuracy_checker/evaluators/custom_evaluators/sr_evaluator.py

```diff
@@ -0,0 +1,572 @@
+        )
+
+    def _create_subset(self, subset=None, num_images=None, allow_pairwise=False):
+        if self.dataset.batch is None:
+            self.dataset.batch = 1
+        if subset is not None:
+            self.dataset.make_subset(ids=subset, accept_pairs=allow_pairwise)
+        elif num_images is not None:
+            self.dataset.make_subset(end=num_images, accept_pairs=allow_pairwise)
+
+    @staticmethod
+    def configure_intermediate_metrics_results(config):
+        compute_intermediate_metric_res = config.get('intermediate_metrics_results', False)
+        metric_interval, ignore_results_formatting = None, None
+        if compute_intermediate_metric_res:
+            metric_interval = config.get('metrics_interval', 1000)
+            ignore_results_formatting = config.get('ignore_results_formatting', False)
+        return compute_intermediate_metric_res, metric_interval, ignore_results_formatting
+
+    def load_network(self, network=None):
+        self.model.load_network(network, self.launcher)
+
+    def load_network_from_ir(self, models_list):
+        self.model.load_model(models_list, self.launcher)
+
+    def get_network(self):
+        return self.model.get_network()
+
+    def get_metrics_attributes(self):
+        if not self.metric_executor:
+            return {}
+        return self.metric_executor.get_metrics_attributes()
+
+    def register_metric(self, metric_config):
+        if isinstance(metric_config, str):
+            self.metric_executor.register_metric({'type': metric_config})
+        elif isinstance(metric_config, dict):
+            self.metric_executor.register_metric(metric_config)
+        else:
+            raise ValueError('Unsupported metric configuration type {}'.format(type(metric_config)))
+
+    def register_postprocessor(self, postprocessing_config):
+        pass
+
+    def register_dumped_annotations(self):
+        pass
+
+    def select_dataset(self, dataset_tag):
+        if self.dataset is not None and isinstance(self.dataset_config, list):
+            return
+        dataset_attributes = create_dataset_attributes(self.dataset_config, dataset_tag)
+        self.dataset, self.metric_executor, self.preprocessor, self.postprocessor = dataset_attributes
+
+    @staticmethod
+    def _create_progress_reporter(check_progress, dataset_size):
+        pr_kwargs = {}
+        if isinstance(check_progress, int) and not isinstance(check_progress, bool):
+            pr_kwargs = {"print_interval": check_progress}
+
+        return ProgressReporter.provide('print', dataset_size, **pr_kwargs)
+
+
+class BaseModel:
+    def __init__(self, network_info, launcher, delayed_model_loading=False):
+        self.network_info = network_info
+        self.launcher = launcher
+
+    def predict(self, idenitifiers, input_data):
+        raise NotImplementedError
+
+    def release(self):
+        pass
+
+# pylint: disable=E0203
+class BaseDLSDKModel:
+    def _reshape_input(self, input_shapes):
```

Comment:
do you use reshape? I assume this model always expect frames with fixed height and width (in other words,  how you can process feedback if all frames have different size?)

Reply:
fixed